### PR TITLE
[HTTP/3] Fix #53632 by using original host when omitted

### LINKED ISSUE #53632: [HTTP/3] UriFormatException when Alt-Svc omits host
### Description
The `Alt-Svc` header from a HTTP/3-enabled site may look like `h3-27=":443"; ma=86400, h3-28=":443"; ma=86400, h3-29=":443"; ma=86400, h3=":443"; ma=86400`, without specifying the host.
In this case, the original host should be assumed. However, `HttpConnectionPool` simply passes `null` to the `HttpAuthority` constructor which directly feeds it into a `UriBuilder`, resulting in a `UriFormatException`.
It should use the same host of the original HTTP request when omitted in `Alt-Svc`.

### Regression?

It would work before 6f6c94dbb2259deb083eefb755698cc8e0f0a4ce but the point is to properly support HTTP/3, which was draft 34 when this issue was filed.

For a temporary workaround, add the following line before the HTTP connection pool is created to disable HTTP/3 draft support.
```cs
AppContext.SetSwitch("System.Net.SocketsHttpHandler.Http3DraftSupport", false);
```

### Other information

Since initial HTTP/3 support:
https://github.com/dotnet/runtime/blob/6f6c94dbb2259deb083eefb755698cc8e0f0a4ce/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs#L800
When this issue was filed:
https://github.com/dotnet/runtime/blob/ad6a69735d61a1bc22da9aff15ebc5df94f63beb/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs#L1007
Where `UriFormatException` is thrown:
https://github.com/dotnet/runtime/blob/ad6a69735d61a1bc22da9aff15ebc5df94f63beb/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpAuthority.cs#L23

### PR DESCRIPTION
**Pull request by wegylexy**
When host is omitted in `Alt-Svc`, use original host with the specified port.

edit:
Fixes #53632

### CODE REVIEW

**Comment by wegylexy** [comment]
In the raw log of the 2 failing checks: `Failed to install tool package 'microsoft.dotnet.xharness.cli': Directory not empty`
Do they have anything to do with `System.Net.Http`?

**Comment by ManickaP** [comment]
Hi @wegylexy,
Thanks for your contribution!
The failures are not related to Http. Also this change might be worth adding a test in https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.AltSvc.cs

**Comment by wegylexy** [comment]
@ManickaP I wonder if all those loopback servers in the unit test share the same host.

**Comment by ManickaP** [comment]
>  I wonder if all those loopback servers in the unit test share the same host.

Yep, they're all localhost.

**Comment by ManickaP** [comment]
Hi @wegylexy, I noticed our tests for AltSvc were disabled, so I put up a PR with a fix. Once, it gets merge, I'll get back to your PR, rerun the tests and we should be good to merge.
So sorry for the hold up and thanks for your contribution!

**Comment by ManickaP** [comment]
So I did a manual confirmation of the change, the other PR is taking too long. It works as expected. So we can merge this.

**Review by ManickaP — APPROVED** [review]

**Pull request merged**
